From 76ae61ef637d7c0faafea8faf98bb0483bd9bbcd Mon Sep 17 00:00:00 2001
From: Francois-Xavier Coudert <fxcoudert@gcc.gnu.org>
Date: Sun, 6 Sep 2020 18:33:04 +0200
Subject: [PATCH] fortran: Fix function arg types for class objects

gcc/fortran/ChangeLog

	* trans-types.c (gfc_get_ppc_type): Fix function arg types.

(cherry picked from commit 7c72651a937533cb67360a6faca05e5603a65865)
Signed-off-by: Kirill A. Korinsky <kirill@korins.ky>
---
 gcc/fortran/trans-types.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git gcc/fortran/trans-types.c gcc/fortran/trans-types.c
index d9f54d85338..6e8d52c3f84 100644
--- gcc/fortran/trans-types.c
+++ gcc/fortran/trans-types.c
@@ -2435,7 +2435,7 @@ gfc_get_ppc_type (gfc_component* c)
   else
     t = void_type_node;
 
-  return build_pointer_type (build_function_type_list (t, NULL_TREE));
+  return build_pointer_type (build_function_type (t, NULL_TREE));
 }
 
 
-- 
2.40.1

